src/js/providers/tracks-mixin.js

```diff
@@ -295,6 +295,7 @@ define(['../utils/underscore',
         this._unknownCount = 0;
         this._activeCuePosition = null;
         if (this._renderNatively) {
+            this.removeTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
```

Comment:
I'd leave a comment here to make it clearer why we have to remove the listener before clearing the cues.